### PR TITLE
Mock server data setup

### LINKED ISSUE #71: Implement mock server api to handle most cases and prevent adding images to assets

### PR DESCRIPTION
**Pull request by amarchenko25**
closed #71

### CODE REVIEW

File: ecommers/lib/core/models/category_model.dart [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  @JsonKey(name: 'subCategory')
+  final List<String> subCategories;
+
+  CategoryModel(this.id, this.title, this.subCategories);
+
+  static const fromJsonFactory = _$CategoryModelFromJson;
+
+  Map<String, dynamic> toJson() => _$CategoryModelToJson(this);
+}
+
```
Comment (Platton96) [inline ecommers/lib/core/models/category_model.dart]: remove empty line

File: ecommers/lib/core/models/data_models/product.dart [hunk 2]
```diff
@@ -0,0 +1,51 @@
+
```
Comment (Platton96) [inline ecommers/lib/core/models/data_models/product.dart]: remove empty line

File: ecommers/lib/core/models/data_models/product_model.dart [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'product_model.g.dart';
+
+@JsonSerializable()
+class ProductModel {
+  final String size;
+  final String color;
+  final int skuId;
```
Comment (Platton96) [inline ecommers/lib/core/models/data_models/product_model.dart]: what is it mean?

File: ecommers/lib/core/provider_models/home_provider_model.dart [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import 'package:ecommers/core/models/data_models/index.dart';
+import 'package:ecommers/core/provider_models/index.dart';
+import 'package:ecommers/core/services/dependency_service.dart';
+import 'package:flutter/material.dart';
+
+class HomeProviderModel extends ProviderModelBase {
+  List<Category> categoryList;
```
Comment (Platton96) [inline ecommers/lib/core/provider_models/home_provider_model.dart]: maybe make this properties get if other class don't change value

File: ecommers/lib/core/services/extensions/get_it_extension.dart [hunk 5]
```diff
@@ -1,8 +1,12 @@
+import 'dart:io';
+
 import 'package:chopper/chopper.dart';
 import 'package:ecommers/core/common/json_serializable_converter.dart';
+import 'package:ecommers/core/models/data_models/index.dart';
 import 'package:ecommers/core/models/index.dart';
 import 'package:ecommers/core/models/login_model.dart';
 import 'package:ecommers/core/services/api_service.dart';
+import 'package:ecommers/core/services/dependency_service.dart';
```
Comment (Platton96) [inline ecommers/lib/core/services/extensions/get_it_extension.dart]: why do you use index

File: ecommers/lib/ui/utils/formatter.dart [hunk 6]
```diff
@@ -6,15 +6,22 @@ class Formatter {
...
+    return 'Master Card ending **$visibleSymbols'; //TODO: use localization
   }
+
   static String getTextWithSpecifiedDateFormat(DateTime date, String format)
   {
     return DateFormat(format).format(date);
   }
+
+  static String getRate(double rate)
+  {
```
Comment (Platton96) [inline ecommers/lib/ui/utils/formatter.dart]: it is dart, incorrect alignment {

File: ecommers/lib/ui/utils/formatter.dart [hunk 7]
```diff
@@ -6,15 +6,22 @@ class Formatter {
...
-    return 'Master Card ending **$visibleSymbols';
+    return 'Master Card ending **$visibleSymbols'; //TODO: use localization
   }
+
   static String getTextWithSpecifiedDateFormat(DateTime date, String format)
   {
     return DateFormat(format).format(date);
   }
+
+  static String getRate(double rate)
```
Comment (Platton96) [inline ecommers/lib/ui/utils/formatter.dart]: param is rate and method getRate maybe you need to rename param or method

File: ecommers/lib/web_server/request_handler.dart [hunk 8]
```diff
@@ -23,21 +38,49 @@ class RequestHandler {
...
+      case ApiDefines.profilePendingPayment:
+      case ApiDefines.profilePendingShipment:
       default:
         _handleUnsupportedRequest(body);
     }
   }
 
-  Future _handleLoginRequest(HttpRequestBody body) async {
-    const String jsonFile = 'login.json';
+  bool isNotAuthorized(HttpRequest request) {
```
Comment (Platton96) [inline ecommers/lib/web_server/request_handler.dart:63]: let's discuss it

File: ecommers/lib/web_server/request_handler.dart [hunk 9]
```diff
@@ -74,36 +115,128 @@ class RequestHandler {
...
+    final searchQuery = queryParameters[ApiQueryParams.searchQuery];
+    final sortType = queryParameters[ApiQueryParams.sortType];
+
+    if (category.isNotNullOrEmpty) {
+      resultProducts = resultProducts.where((p) => p.category == category);
+    } else if (subCategory.isNotNullOrEmpty) {
+      resultProducts =
+          resultProducts.where((p) => p.subCategory == subCategory);
+    } else if (rangeFrom.isNotNullOrEmpty && rangeTo.isNotNullOrEmpty) {
+      resultProducts =
```
Comment (Platton96) [inline ecommers/lib/web_server/request_handler.dart:138]: too lot of 'if'

File: ecommers/lib/web_server/request_handler.dart [hunk 10]
```diff
@@ -1,14 +1,29 @@
...
+
+import 'package:ecommers/core/common/api_query_params.dart';
+import 'package:ecommers/core/models/data_models/index.dart';
+import 'package:ecommers/core/models/sort_type.dart';
+import 'package:ecommers/web_server/data_access/products_data_access.dart';
+import 'package:ecommers/web_server/services/product_comparator.dart';
+import 'package:enum_to_string/enum_to_string.dart';
 import 'package:http_server/http_server.dart';
+import '../core/common/index.dart';
+import '../extensions/string_extension.dart';
```
Comment (Platton96) [inline ecommers/lib/web_server/request_handler.dart:13]: change to full path

File: ecommers/lib/web_server/request_handler.dart (hunk 9 above)
Comment (Platton96) [reply ecommers/lib/web_server/request_handler.dart:138]: maybe make special method for creating resultProducts

File: ecommers/lib/core/services/extensions/get_it_extension.dart [hunk 11]
```diff
@@ -1,22 +1,44 @@
...
-import 'package:ecommers/core/common/json_serializable_converter.dart';
-import 'package:ecommers/core/models/index.dart';
-import 'package:ecommers/core/models/login_model.dart';
-import 'package:ecommers/core/services/api_service.dart';
-import 'package:ecommers/web_server/local_server.dart';
 import 'package:get_it/get_it.dart';
 
+import '../../../web_server/local_server.dart';
+import '../../common/index.dart';
+import '../../models/data_models/index.dart';
```
Comment (Platton96) [inline ecommers/lib/core/services/extensions/get_it_extension.dart:8]: use full path

Comment (amarchenko25) [reply ecommers/lib/core/services/extensions/get_it_extension.dart:8]: according to style guides, we shouldn't use full path

**Review by Platton96 — APPROVED** [review]

**Pull request merged**